packages/cli/src/splitRunner/splitRunner.ts

```diff
@@ -1,231 +1,44 @@
-
-        imports.forEach((depImport) => {
-          if (depImport.isExternal || !depImport.source) {
-            // Ignore external dependencies
-            return;
-          }
-
-          // for each import, reconstruct the export map
-          const depExport = exportMap.get(depImport.source);
-          if (!depExport) {
-            throw new Error("Export not found");
-          }
-
-          // check named imports
-        });
-
-        return file;
-      });
-    }
-    // TODO
-    // Step 1, create variable to track which export is used
-    // Step 2, iterate over all file imports. If the import is used, mark the export as used
-    // Step 3, iterate over each file, and remove the unused exports
-
-    // Repeat above step until no more unused exports are found
-    assert(exportMap);
-  }
-
-  #removeErrors() {
-    this.files = this.files.map((file) => {
-      const languagePlugin = getLanguagePlugin(this.entrypointPath, file.path);
-
-      const tree = languagePlugin.parser.parse(file.sourceCode);
-
-      const indexesToRemove: { startIndex: number; endIndex: number }[] = [];
-
-      const query = new Parser.Query(
-        languagePlugin.parser.getLanguage(),
-        "(ERROR) @error",
-      );
-      const errorCaptures = query.captures(tree.rootNode);
-      errorCaptures.forEach((capture) => {
-        indexesToRemove.push({
-          startIndex: capture.node.startIndex,
-          endIndex: capture.node.endIndex,
-        });
+      worker.on("error", reject);
+      worker.on("exit", (code) => {
+        if (code !== 0) {
+          reject(new Error(`Worker stopped with exit code ${code}`));
+        }
       });
-
-      const updatedSourceCode = removeIndexesFromSourceCode(
-        file.sourceCode,
-        indexesToRemove,
-      );
-
-      return { ...file, sourceCode: updatedSourceCode };
     });
   }
-
-  run() {
-    console.info("\n");
-    console.time("Splitting");
-
-    console.time("remove annotation from other groups");
-    this.#removeAnnotationFromOtherGroups();
-    console.timeEnd("remove annotation from other groups");
-
-    console.time("Get export map");
-    const exportMap = this.#getExportMap();
-    console.timeEnd("Get export map");
-
-    console.time("Remove invalid imports and usages");
-    this.#removeInvalidImportsAndUsages(exportMap);
-    console.timeEnd("Remove invalid imports and usages");
-
-    console.time("Remove unused imports");
-    this.#removeUnusedImports();
-    console.timeEnd("Remove unused imports");
-
-    console.time("Remove unused files");
-    this.#removeUnusedFiles();
-    console.timeEnd("Remove unused files");
-
-    console.time("Remove unused exports");
-    this.#removeUnusedExports(exportMap);
-    console.timeEnd("Remove unused exports");
-
-    console.time("Remove errors");
-    this.#removeErrors();
-    console.timeEnd("Remove errors");
-
-    console.timeEnd("Splitting");
```

Comment:
Need to find a better way to log. Maybe with a fancier logger. Cause here since they now run in parallel, very hard to read the logs